controllers/manifest_test.go

```diff
@@ -61,5 +77,299 @@ var _ = Describe("Update Manifest CR", Ordered, func() {
 
 		By("CR updated with new value in spec.resource.spec")
 		Eventually(expectManifestSpecDataEquals(kyma.Name, valueUpdated), Timeout, Interval).Should(Succeed())
+
+		By("Update Module Template spec.descriptor.component values")
+		{
+			newComponentDescriptorRepositoryURL := func(moduleTemplate *v1beta2.ModuleTemplate) error {
+				descriptor, err := moduleTemplate.GetDescriptor()
+				if err != nil {
+					return err
+				}
+
+				repositoryContext := descriptor.GetEffectiveRepositoryContext().Object
+				_, ok := repositoryContext["baseUrl"].(string)
+				if !ok {
+					Fail("Can't find \"baseUrl\" property in ModuleTemplate spec")
+				}
+				repositoryContext["baseUrl"] = updateRepositoryURL
+
+				newDescriptorRaw, err := compdesc.Encode(descriptor.ComponentDescriptor, compdesc.DefaultJSONLCodec)
+				Expect(err).ToNot(HaveOccurred())
+				moduleTemplate.Spec.Descriptor.Raw = newDescriptorRaw
+
+				return nil
+			}
+			updateKCPModuleTemplateWith := updateKCPModuleTemplate("skr-module-update", "default")
+
+			Eventually(updateKCPModuleTemplateWith(newComponentDescriptorRepositoryURL), Timeout, Interval).Should(Succeed())
+		}
+
+		By("Manifest is updated with new value in spec.install.source")
+		{
+			hasDummyRepositoryURL := func(manifest *v1beta2.Manifest) error {
+				manifestImageSpec := extractInstallImageSpec(manifest.Spec.Install)
+				if !strings.HasPrefix(manifestImageSpec.Repo, updateRepositoryURL) {
+					return fmt.Errorf("Invalid manifest spec.install.repo: %s, expected prefix: %s", //nolint:goerr113
+						manifestImageSpec.Repo, updateRepositoryURL)
+				}
+				return nil
+			}
+
+			expectManifest := expectManifestFor(kyma)
+			Eventually(expectManifest(hasDummyRepositoryURL), Timeout, Interval).Should(Succeed())
+		}
+	})
+})
+
+var _ = Describe("Manifest.Spec is rendered correctly", Ordered, func() {
+	kyma := NewTestKyma("kyma")
+
+	module := v1beta2.Module{
+		ControllerName: "manifest",
+		Name:           NewUniqModuleName(),
+		Channel:        v1beta2.DefaultChannel,
+	}
+	kyma.Spec.Modules = append(kyma.Spec.Modules, module)
+	RegisterDefaultLifecycleForKyma(kyma)
+
+	It("validate Manifest", func() {
+		moduleTemplate, err := GetModuleTemplate(ctx, controlPlaneClient, module.Name, "default")
```

Comment:
Please verify that the `sync-oci-ref` annotation also exists in this test

Reply:
This annotation can't be tested here, because it's set by the Manifest controller, which is disabled in this test.
I'll test it's presence in the subsequent PR for tests in the `internal/manifest` package.